### ISSUE TITLE
Accessibility Standards

### ISSUE THREAD

**Issue by asantaellaSB** [understrap/understrap#1081]
## Issue is about..
  - [ X ] a Suggestion
  - [ X ] a Question
---

I've been asked to provide a statement of accessibility for a site using the Understrap theme. I know that the theme can be altered and can be made inaccessible, but is there any page or language I can use to state that the initial theme used "Understrap" is accessible to certain ADA standards? If not, can you provide the standards the site was developed against?

**Comment by Thomas-A-Reinert** [comment]
Hi @asantaellaSB - guess it would be best to get a dedicated tester to test it. We cannot _guarantee_ 100% accessibility against WCAG 2.1 AAA standard - or whatever standard you need to comply to. A? AA? AAA?

But maybe it´s a good idea setting up such a statement. Anyway as it´s an open source project, it could easily fail the standard with every commit someone contributes.

There is a whole heap of tools available to check for WCAG compliance, already built into Chrome and Firefox too. Also via online-checkers etc. like https://achecker.ca/checker/index.php but even if you seem to pass there, that doesn´t really mean you do fully comply. A lot of tests can really only be done by people, not machines/software.

For example: my own site gets 99% points for accessibility in Chrome audit. 1% off for having a duplicate ID (damn stupid, need to correct that ;) ) but I KNOW it fails an a whole lot of different other issues like contrast etc.

**Comment by Thomas-A-Reinert** [comment]
I´m currently getting trained to become a certified WCAG tester/dev. So chances are good that I´ll review the code of the parent theme anytime soon. But as you already mentioned - Accessibility is a concept as a whole that doesn´t only affect the code but furthermore graphics, contrast, readability, texting and so much more.

Chances are high that your customer/content editor f****s up a sites compliance with just a single post when he doesn´t know about how to write and what to take into consideration adding images etc.

**Comment by pattonwebz** [comment]
The base theme passes many of the AA guidelines but not them all. I don't recall exactly which ones but when I checked it in the past against the absolutely minimal WP.org A11Y tag guidelines there was something that didn't pass.

**Comment by axlright** [comment]
A lot of WCAG accessibility into the AA and AAA has a lot to do with color choice, so the theme won't impact that. I'm working on a WCAG AA project now and will keep note of any parts of the base theme that may need modification. Thanks!

**Comment by pattonwebz** [comment]
Tap target size and index (tabbing) order I think were what didn't pass in my checks but I really can't remember for sure. Plenty of improvements to be made I am sure if someone gives it a full audit :)

**Comment by Thomas-A-Reinert** [comment]
Darn. Wrote that thing up once already and then – page reloaded (for whatever reason) and all got lost :/

@axlright : my goal is to rewrite that theme to make it comply to WCAG 2.1 standards AA+ level from stock. Including colors etc. as far as possible.
Yes, you´re right – that can easily be compromised. But that´s up to the dev who wants to customize it. Not our problem though.

@pattonwebz : Yes, automated tools/checkers don´t give you the full perspective especially concerning document outline, aria-roles and labels.
That can only be done by humans. But you should comply to the testing tools suggestions first and THEN get into human testing, which would cut costs short ;)

@asantaellaSB : Well, I totally understand. And I think it´d be a great chance for UnderStrap to extend it´s market share once it complies to WCAG 2.1/ADA standards as far as possible. And that this can be confirmed and be officially communicated. That´s why I currently feel enforced to just do that as you can see on #1088 .

But please be aware:
* This is work in progress and really, REALLY not meant to be put into production **YET**!
* There´s a whole lot that still needs improvement, like the afore mentioned document outline as a whole. And I´m only half-way through.
  * AND: It will f**k up most users existing document outline.
  * I guess that won´t be critical for a "food blog" for example, but it may impact SEO as well as well as other things. Just a sidenote.
  * WE NEED INSTRUCTIONS ON HOW TO HANDLE UnderStrap CORRECTLY, @UnderstrapFramework 
    * That being said your focus should be documentation of theme UnderStrap. Check my repo/site for that. We got an updated version avail compared to the one that "UnderStrap" sports. It still needs updates though. Which got neglected due to the fact that we never received any feedback from holger.
    * But WCAG-compliance would make out another important section – if anyone care to fill that thing.


@asantaellaSB - Even if we totally comply to my official tester: That´s a personal agreement. She won´t put her name under it nor will it have an official "BIK-approved" label. That one will cost about 1.600€, so I guess you know know why an open source project won´t have that. In case you want to sponsor an official test - you´re more than welcome!

**Comment by Thomas-A-Reinert** [comment]
As the question has been adequately been answered I guess, I´m going to close this non-issue.

Feel free to re-open it if any question remain open, @asantaellaSB asantaellaSB
